analyzers/analysistest/analysistest.go

```diff
@@ -0,0 +1,142 @@
+package analysistest
+
+import (
+	"context"
+	"errors"
+	"os"
+	"path/filepath"
+	"regexp"
+	"sort"
+	"strings"
+
+	"github.com/deepsourcelabs/deepsource-go/analyzers/types"
+	sitter "github.com/smacker/go-tree-sitter"
+	"github.com/smacker/go-tree-sitter/css"
+	"github.com/smacker/go-tree-sitter/golang"
+)
+
+// ParsedIssue represents an issue parsed using tree-sitter.
+type ParsedIssue struct {
+	IssueCode string
+	Line      int
+}
+
+// Verify compares the generated report and parsed issues using tree-sitter.
+func Verify(report types.AnalysisReport, filename string) error {
```

Comment:
Can the developer pass in the file type as well? This will help if there are certain files that can't be figured out with just the extension (like Dockerfiles).

This can be part of a subsequent version.

Reply:
Ref: #4.